src/Core/Managed/Net45/Extensibility/Implementation/RichPayloadEventSource.TelemetryHandler.cs

```diff
@@ -0,0 +1,604 @@
+
+                // Trace
+                this.telemetryHandlers.Add(typeof(TraceTelemetry), this.CreateHandlerForTraceTelemetry(writeGenericMethod, eventSourceOptionsType, eventSourceOptionsKeywordsProperty));
+
+                // Event
+                this.telemetryHandlers.Add(typeof(EventTelemetry), this.CreateHandlerForEventTelemetry(writeGenericMethod, eventSourceOptionsType, eventSourceOptionsKeywordsProperty));
+
+                // Dependency
+                this.telemetryHandlers.Add(typeof(DependencyTelemetry), this.CreateHandlerForDependencyTelemetry(writeGenericMethod, eventSourceOptionsType, eventSourceOptionsKeywordsProperty));
+
+                // Metric
+                this.telemetryHandlers.Add(typeof(MetricTelemetry), this.CreateHandlerForMetricTelemetry(writeGenericMethod, eventSourceOptionsType, eventSourceOptionsKeywordsProperty));
+
+                // Exception
+                this.telemetryHandlers.Add(typeof(ExceptionTelemetry), this.CreateHandlerForExceptionTelemetry(writeGenericMethod, eventSourceOptionsType, eventSourceOptionsKeywordsProperty));
+
+                // PerformanceCounter
+                this.telemetryHandlers.Add(typeof(PerformanceCounterTelemetry), this.CreateHandlerForPerformanceCounterTelemetry(writeGenericMethod, eventSourceOptionsType, eventSourceOptionsKeywordsProperty));
+
+                // PageView
+                this.telemetryHandlers.Add(typeof(PageViewTelemetry), this.CreateHandlerForPageViewTelemetry(writeGenericMethod, eventSourceOptionsType, eventSourceOptionsKeywordsProperty));
+
+                // SessionState
+                this.telemetryHandlers.Add(typeof(SessionStateTelemetry), this.CreateHandlerForSessionStateTelemetry(writeGenericMethod, eventSourceOptionsType, eventSourceOptionsKeywordsProperty));
+            }
+            else
+            {
+                CoreEventSource.Log.LogVerbose("Unable to get method: EventSource.Write<T>(String, EventSourceOptions, T)");
+            }
+        }
+
+        private Action<ITelemetry> CreateHandlerForRequestTelemetry(MethodInfo writeGenericMethod, Type eventSourceOptionsType, PropertyInfo eventSourceOptionsKeywordsProperty)
+        {
+            var eventSourceOptions = Activator.CreateInstance(eventSourceOptionsType);
+            var keywords = Keywords.Requests;
+            eventSourceOptionsKeywordsProperty.SetValue(eventSourceOptions, keywords);
+            var dummyRequestData = new RequestData();
+            var writeMethod = writeGenericMethod.MakeGenericMethod(new
+            {
+                PartA_iKey = this.dummyPartAiKeyValue,
+                PartA_Tags = this.dummyPartATagsValue,
+                PartB_RequestData = new
+                {
+                    // The properties and layout should be the same as RequestData_types.cs
```

Comment:
That's the reason why we need anonymous type. To use the code like "PartB_RequestData = new RequestData". RequestData needs to have EventDataAttribute but it's only available since .Net 4.6.